### ISSUE TITLE
"'bool' is not a valid base class" should be possible to ignore.

### ISSUE THREAD

**Issue by wookie184** [python/mypy#13411]
**Bug Report**

This code runs without error in Python, but mypy gives an error: 
```python
class Meta(type):
    def __new__(cls, name, bases, dct):
        return super().__new__(cls, name, (), dct)
    
class Foo(bool, metaclass=Meta):  # type: ignore
    ...

x = Foo()
```

The error is:
```python
error: "bool" is not a valid base class
Found 1 error in 1 file (errors prevented further checking)
```
Without the `type: ignore` it makes sense that an error is raised, but given that it's possible that it is valid code, it would be nice if it could be ignored.

Pylance seems to handle the above code without error.

**Use Case**
I came across this whilst looking into ORM code with 3.11's `dataclass_transform`. Example code is here:
https://mypy-play.net/?mypy=master&python=3.11&gist=763ad45ab009be9e0f7337347a299512

If there's a different way of getting the code shown to work I'd be interested to know, all I could think of was making `BoolField` a function instead of a class.

**Your Environment**

- Mypy version used: main (mypy 0.980+dev.b8055512bd3cc54a0b82f8382ac39b3430cf9941.dirty (compiled: no))
- Mypy command-line flags: None
- Mypy configuration options from `mypy.ini` (and other config files): None
- Python version used: 3.10.1

Thanks for mypy :)

**Comment by wookie184** [comment]
Looked into  implementing this and I see two main options:
1) Remove the check in the semantic analyser completely and let it fall back to the check in the type checker: `Cannot inherit from final class "bool"`
2) Make the check in the semantic analyser not a blocker, and make the type checker not raise the "final class" error for `bool`s (to prevent duplicate errors).

Assuming you'd be happy for this change, would either be preferable?

**Comment by hauntsaninja** [comment]
Thanks for bringing this up, I would merge a PR for option 1 (fall back to final check)

It looks like this was made a blocking check way back when in https://github.com/python/mypy/pull/1834. And the actual check goes back to 2013! Anyway, back then mypy had a lot more blocking checks and final did not exist.
